tests/test_converters.py

```diff
@@ -91,3 +92,76 @@ def test_valid_python_identifier_for_valid(value: str):
+        # Simple duration strings
+        ('1Y', datetime.datetime.fromisoformat('2020-01-01T00:00:00')),
+        ('1y', datetime.datetime.fromisoformat('2020-01-01T00:00:00')),
+        ('1year', datetime.datetime.fromisoformat('2020-01-01T00:00:00')),
+        ('1years', datetime.datetime.fromisoformat('2020-01-01T00:00:00')),
+        ('1m', datetime.datetime.fromisoformat('2019-02-01T00:00:00')),
+        ('1month', datetime.datetime.fromisoformat('2019-02-01T00:00:00')),
+        ('1months', datetime.datetime.fromisoformat('2019-02-01T00:00:00')),
+        ('1w', datetime.datetime.fromisoformat('2019-01-08T00:00:00')),
+        ('1W', datetime.datetime.fromisoformat('2019-01-08T00:00:00')),
+        ('1week', datetime.datetime.fromisoformat('2019-01-08T00:00:00')),
+        ('1weeks', datetime.datetime.fromisoformat('2019-01-08T00:00:00')),
+        ('1d', datetime.datetime.fromisoformat('2019-01-02T00:00:00')),
+        ('1D', datetime.datetime.fromisoformat('2019-01-02T00:00:00')),
+        ('1day', datetime.datetime.fromisoformat('2019-01-02T00:00:00')),
+        ('1days', datetime.datetime.fromisoformat('2019-01-02T00:00:00')),
+        ('1h', datetime.datetime.fromisoformat('2019-01-01T01:00:00')),
+        ('1H', datetime.datetime.fromisoformat('2019-01-01T01:00:00')),
+        ('1hour', datetime.datetime.fromisoformat('2019-01-01T01:00:00')),
+        ('1hours', datetime.datetime.fromisoformat('2019-01-01T01:00:00')),
+        ('1M', datetime.datetime.fromisoformat('2019-01-01T00:01:00')),
+        ('1minute', datetime.datetime.fromisoformat('2019-01-01T00:01:00')),
+        ('1minutes', datetime.datetime.fromisoformat('2019-01-01T00:01:00')),
+        ('1s', datetime.datetime.fromisoformat('2019-01-01T00:00:01')),
+        ('1S', datetime.datetime.fromisoformat('2019-01-01T00:00:01')),
+        ('1second', datetime.datetime.fromisoformat('2019-01-01T00:00:01')),
+        ('1seconds', datetime.datetime.fromisoformat('2019-01-01T00:00:01')),
+
+        # Complex duration strings
+        ('1y1m1w1d1H1M1S', datetime.datetime.fromisoformat('2020-02-09T01:01:01')),
+        ('5y100S', datetime.datetime.fromisoformat('2024-01-01T00:01:40')),
+        ('2w28H', datetime.datetime.fromisoformat('2019-01-16T04:00:00')),
+    )
+)
+def test_duration_converter_for_valid(duration: str, expected: datetime):
+    converter = Duration()
+
+    with patch('bot.converters.datetime') as mock_datetime:
+        mock_datetime.utcnow.return_value = FIXED_UTC_NOW
+        assert asyncio.run(converter.convert(None, duration)) == expected
+
+
+@pytest.mark.parametrize(
+    ('duration'),
+    (
+        # Units in wrong order
+        ('1d1w'),
+        ('1s1y'),
+
+        # Unknown substrings
+        ('1MVes'),
+        ('1y3breads'),
+
+        # Missing amount
+        ('ym'),
+
+        # Garbage
+        ('Guido van Rossum'),
+        ('lemon lemon lemon lemon lemon lemon lemon'),
+    )
+)
+def test_duration_converter_for_invalid(duration: str):
```

Comment:
Some tests for duplicate units would be good

Reply:
I'll add that.